internal/handlers/common/update.go

```diff
@@ -41,6 +42,66 @@ func UpdateDocument(doc, update *types.Document) error {
 				}
 			}
 
+		case "$inc":
+			incDoc, ok := updateV.(*types.Document)
```

Comment:
Why don't we move that long construction to a separate function? So then it's easier to test and the cyclomatic complexity is a bit lower (and it's easier to read it). I don't want to block this PR merge by it, but I believe we should discuss this topic.

Reply:
There would be no test impact if I move it to a separate function as we test with integration tests mostly.
I believe it's just because we don't have an agreement on when it's time to extract a function right now.